### PR TITLE
T12:Event Activity Scaffold

### PR DESCRIPTION
**Pull request by Animansoubi**
Add button navigation.
Add Home/Profile/my event Fragments

### CODE REVIEW

File: app/src/main/java/com/gathering/android/NavHostActivity.kt [hunk 1]
```diff
@@ -2,18 +2,39 @@ package com.gathering.android
...
 import dagger.hilt.android.AndroidEntryPoint
+import javax.inject.Inject
 
 @AndroidEntryPoint
 class NavHostActivity : AppCompatActivity() {
 
     private lateinit var binding: ActNaveHostBinding
 
+    @Inject
+    lateinit var authRepository: AuthRepository
```
Comment (moyarsuccess) [inline app/src/main/java/com/gathering/android/NavHostActivity.kt:19]: You should not have a repository in the view layer directly, you need to have a VM here.

File: app/src/main/java/com/gathering/android/intro/IntroFragment.kt [hunk 2]
```diff
@@ -1,29 +1,42 @@
...
-class IntroFragment : Fragment() {
+class IntroFragment : DialogFragment() {
 
     private lateinit var binding: FrgIntroBinding
 
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setStyle(
+            STYLE_NORMAL,
+            android.R.style.Theme_Black_NoTitleBar_Fullscreen
```
Comment (moyarsuccess) [inline app/src/main/java/com/gathering/android/intro/IntroFragment.kt:23]: Do an investigation to see how to change the theme to something light, not dark.

File: app/src/main/java/com/gathering/android/signin/SignInFragment.kt [hunk 3]
```diff
@@ -4,20 +4,29 @@ import android.os.Bundle
...
-class SignInFragment : Fragment() {
+class SignInFragment : DialogFragment() {
 
     private lateinit var binding: FrgSignInBinding
 
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setStyle(
+            STYLE_NORMAL,
+            android.R.style.Theme_Black_NoTitleBar_Fullscreen
```
Comment (moyarsuccess) [inline app/src/main/java/com/gathering/android/signin/SignInFragment.kt:21]: Same here

File: app/src/main/java/com/gathering/android/signup/SignUpFragment.kt [hunk 4]
```diff
@@ -1,31 +1,38 @@
...
     private lateinit var binding: FrgSignUpBinding
 
     @Inject
     lateinit var viewModel: SignUpViewModel
 
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setStyle(
+            STYLE_NORMAL,
+            android.R.style.Theme_Black_NoTitleBar_Fullscreen
```
Comment (moyarsuccess) [inline app/src/main/java/com/gathering/android/signup/SignUpFragment.kt:27]: Same here

File: app/src/main/java/com/gathering/android/signup/model/AuthRepository.kt [hunk 5]
```diff
@@ -26,6 +26,26 @@ class AuthRepository @Inject constructor() {
             }
     }
 
+    fun signInUser(email: String, pass: String, onResponseReady: (ResponseState) -> Unit) {
```
Comment (moyarsuccess) [inline app/src/main/java/com/gathering/android/signup/model/AuthRepository.kt:29]: This function is going to be added by @idaosk in another PR. You can remove this from here.

File: app/src/main/res/layout/act_nave_host.xml [hunk 6]
```diff
@@ -1,18 +1,32 @@
 <?xml version="1.0" encoding="utf-8"?>
```
Comment (moyarsuccess) [inline app/src/main/res/layout/act_nave_host.xml:1]: The file name should be act_nav_host there is a typo in that.

File: app/src/main/res/menu/bottom_nav_menu.xml [hunk 7]
```diff
@@ -0,0 +1,12 @@
+<?xml version="1.0" encoding="utf-8"?>
+<menu xmlns:android="http://schemas.android.com/apk/res/android">
+    <item
```
Comment (moyarsuccess) [inline app/src/main/res/menu/bottom_nav_menu.xml:3]: Add icons defined above here.

**Review by moyarsuccess — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/gathering/android/NavHostActivity.kt (hunk 1 above)
Comment (Animansoubi) [reply app/src/main/java/com/gathering/android/NavHostActivity.kt:19]: It was solved. I created the ViewModel for that

File: app/src/main/java/com/gathering/android/intro/IntroFragment.kt (hunk 2 above)
Comment (Animansoubi) [reply app/src/main/java/com/gathering/android/intro/IntroFragment.kt:23]: Done!

File: app/src/main/java/com/gathering/android/signin/SignInFragment.kt (hunk 3 above)
Comment (Animansoubi) [reply app/src/main/java/com/gathering/android/signin/SignInFragment.kt:21]: Done!

File: app/src/main/java/com/gathering/android/signup/SignUpFragment.kt (hunk 4 above)
Comment (Animansoubi) [reply app/src/main/java/com/gathering/android/signup/SignUpFragment.kt:27]: Done!

File: app/src/main/java/com/gathering/android/signup/model/AuthRepository.kt (hunk 5 above)
Comment (Animansoubi) [reply app/src/main/java/com/gathering/android/signup/model/AuthRepository.kt:29]: Done!

File: app/src/main/res/layout/act_nave_host.xml (hunk 6 above)
Comment (Animansoubi) [reply app/src/main/res/layout/act_nave_host.xml:1]: Done!

**Review by idaosk — APPROVED** [review]
looks good!

**Review by moyarsuccess — APPROVED** [review]

**Pull request merged**